tabcmd/execution/tabcmd_controller.py

```diff
@@ -23,29 +23,27 @@ def run(parser, user_input=None):
             sys.exit(0)
         user_input = user_input or sys.argv[1:]
         namespace = parser.parse_args(user_input)
-        if namespace.logging_level and namespace.logging_level != logging.INFO:
+        if hasattr("namespace", "logging_level") and namespace.logging_level != logging.INFO:
             print("logging:", namespace.logging_level)
 
         logger = log(__name__, namespace.logging_level or logging.INFO)
-        if namespace.password or namespace.token_value:
-            logger.trace(namespace.func)
+        if (hasattr("namespace", "password") or hasattr("namespace", "token_value")) and hasattr("namespace", "func"):
```

Comment:
Do Python IDEs support the notion of renaming object attributes? If so, would these strings like `"token_value"` get updated too?

Reply:
They do in general, but I don't know any way that would catch the string usages. If we wanted that to work, this line could be changed to 
`try:
    if (namespace.password ...):
except AttributeError:
    # it doesn't exist
`

In this specific case, automated refactoring wouldn't work anyway - the attributes of namespace are defined at runtime as we add potential arguments to the parser in parent_parser/each command definition. So in this line: 
`    secret_values.add_argument( "-p", **"--password"**, default=None, metavar="<PASSWORD>", help=_("session.options.password") )`

the option name **--password** gets turned into the attribute namespace.password.